### ISSUE TITLE
Allow passing a function name to rename_all and rename_all_fields

### ISSUE THREAD

**Issue by njames93** [serde-rs/serde#2852]
Currently `rename_all` is pretty restrictive in what it allows. It would be nice if you could pass the name of a function and use that to rename the fields.

The function should have the signature
```rs
fn rename_field(&str) -> String
```

The specific use case is for tagged enums, where the tag in my case has a bit of preamble at the start of the string that is shared with all tags.
```json
[
  {
    "type": "MyLongPreamble.Foo"
  },
  {
    "type": "MyLongPreamble.Bar"
  }
]
```
In this case it would be nice to have a function like this
```rs
fn remap(val: &str) -> String {
  "MyLongPreamble.".to_owned() + val
}

#[serde(rename_all = "remap")]
pub enum Item {
  Foo,
  Bar
}
```

**Comment by oli-obk** [comment]
I don't think this is doable. These functions would need to be run at derive time, which is a concept rustc does not support yet (unclear if ever).
